tests/run-make/rust-lld-custom-target/rmake.rs

```diff
@@ -0,0 +1,51 @@
+// Test linking using `cc` with `rust-lld`, using a custom target with features described in MCP 510
+// see https://github.com/rust-lang/compiler-team/issues/510 for more info:
+//
+// Starting from the `x86_64-unknown-linux-gnu` target spec, we add the following options:
+// - a linker-flavor using lld via a C compiler
+// - the self-contained linker component is enabled
+
+//@ needs-rust-lld
+//@ only-x86_64-unknown-linux-gnu
+
+extern crate run_make_support;
+
+use run_make_support::regex::Regex;
+use run_make_support::rustc;
+use std::process::Output;
+
+fn main() {
+    // Compile to a custom target spec with rust-lld enabled by default. We'll check that by asking
+    // the linker to display its version number with a link-arg.
+    let output = rustc()
+        .env("RUSTC_LOG", "rustc_codegen_ssa::back::link=info")
+        .arg("--crate-type=cdylib")
+        .arg("--target=custom-target.json")
+        .arg("-Clink-args=-Wl,-v")
+        .input("lib.rs")
+        .run();
+    assert!(
+        find_lld_version_in_logs(output),
+        "the LLD version string should be present in the output logs"
+    );
+
+    // But it can also be disabled via linker features.
+    let output = rustc()
+        .env("RUSTC_LOG", "rustc_codegen_ssa::back::link=info")
+        .arg("--crate-type=cdylib")
+        .arg("--target=custom-target.json")
```

Comment:
This can be

```rs
rustc()
    .crate_type("cdylib")
    .target("custom-target.json")
    // ...
```

Reply:
Done. 

I've left the `fn target` implementation as-is, but it's mentioning target triples only (and ensuring that there are no spaces) which you can see here could be improved to mention json target specs. I'm not sure if that's still unstable though.

Reply:
Could you edit the `target()` docs to mention custom JSON specs (at the time of writing I didn't consider target JSON specs), and remove the assert for whitespace check? As for target JSON existence, rustc would fail on not finding a non-existent target spec anyway.

Reply:
Done